### ISSUE TITLE
Output filename flag will fail unless parent directories exist on Windows

### ISSUE THREAD

**Issue by dunkmann00** [Nuitka/Nuitka#1452]
Nuitka Version:
```
0.7.2
Commercial: None
Python: 3.9.9 (tags/v3.9.9:ccb0e6a, Nov 15 2021, 18:08:50) [MSC v.1929 64 bit (AMD64)]
Flavor: Unknown
Executable: ...\venv\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
```

Installed with `pip` into a virtual environment.

I am building with `--onefile` mode and am specifying an output filename with `-o`. When I do this on Mac it works fine even if parent directories for the output are missing. On Windows this does not work and produces an error that says 'The system cannot find the path specified.'

```python
Traceback (most recent call last):
  File "...\venv\lib\site-packages\nuitka\__main__.py", line 137, in <module>
    main()
  File "...\venv\lib\site-packages\nuitka\__main__.py", line 123, in main
    MainControl.main()
  File "...\venv\lib\site-packages\nuitka\MainControl.py", line 974, in main
    packDistFolderToOnefile(dist_dir, binary_filename)
  File "...\venv\lib\site-packages\nuitka\freezer\Onefile.py", line 72, in packDistFolderToOnefile
    packDistFolderToOnefileBootstrap(onefile_output_filename, dist_dir)
  File "...\venv\lib\site-packages\nuitka\freezer\Onefile.py", line 395, in packDistFolderToOnefileBootstrap
    _runOnefileScons(
  File "...\venv\lib\site-packages\nuitka\freezer\Onefile.py", line 293, in _runOnefileScons
    result = SconsInterface.runScons(
  File "...\venv\lib\site-packages\nuitka\build\SconsInterface.py", line 254, in runScons
    options["result_exe"] = getExternalUsePath(
  File "...\venv\lib\site-packages\nuitka\utils\FileOperations.py", line 759, in getExternalUsePath
    dirname = getWindowsShortPathName(os.path.dirname(filename))
  File "...\venv\lib\site-packages\nuitka\utils\FileOperations.py", line 728, in getWindowsShortPathName
    raise WindowsError(
ProcessLookupError: [Errno 3] The system cannot find the path specified.
```

I'm not sure if you would consider this an error, or if you think the directories should already exist. But either way, the behavior is different on Mac vs Windows, which might be surprising to users (it personally tripped me up).

Thanks for Nuitka!

**Comment by kayhayen** [comment]
The code in question tries to avoid unusable code paths on Windows, but for non-existent paths, it errors out, on macOS there is no such problem. I assume therefore it's trigger on Windows, but not working either way.

I guess, creating the directory is a way to go. In a sense, the `-o` was mostly intended to allow a non-default filename, not necessarily the output directory, but I don't think combining `-o my_name --output-dir=some_dir` combination is currently working, so the path would have to be repeated, or would have to applied.

I tend now to make it an error to specify a path, and to apply the output directory to it, making it so, that you only specify a non-standard name with "-o", which is what it is actually for.

**Comment by dunkmann00** [comment]
Ahh I see, I was misusing the `-o` by giving it a path rather than a name. Interestingly enough, doing so actually works perfectly fine on macOS (what I mean by that is it will create the directories that don't exist if necessary and the binary that it produces runs just fine).

I also was able to successfully combine `-o arbitrary/path/to/my_name --output-dir=some_dir` and get that to work (on both Windows and macOS), so long as the directories in the `-o` path exist.

However, since that is not the intended way to use `-o` I understand why I was having problems and I think your solution to explicitly make it an error makes sense.

Thanks!

**Comment by kayhayen** [comment]
```
Nuitka-Options: Used command line options: --onefile --standalone --company-name="Nuitka Test Company"
Nuitka-Options: --product-name="Hello World Deluxe" --product-version=1 --output-file=d/d
Nuitka-Options: tests/onefile/HelloWorldTest.py
FATAL: Error, output filename for standalone cannot contain a directory part.
```

Thanks to @KevinRodriguez777 for pointing out that this was solved a while ago, the current 2.0 rejects this usage, for `--output-dir` it will create things, but the `--output-filename` is no more letting these things slip by, combine them if you want.
